opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/validation/AccountAccessBodyValidator.java

```diff
@@ -0,0 +1,63 @@
+package de.adorsys.opba.protocol.xs2a.service.xs2a.validation;
+
+import de.adorsys.opba.protocol.xs2a.service.xs2a.annotations.ValidConsentBody;
+import de.adorsys.opba.protocol.xs2a.service.xs2a.dto.consent.AisConsentInitiateBody;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import static de.adorsys.opba.protocol.xs2a.service.xs2a.dto.consent.AccountAccessType.ALL_ACCOUNTS;
+import static de.adorsys.opba.protocol.xs2a.service.xs2a.dto.consent.AccountAccessType.ALL_ACCOUNTS_WITH_BALANCES;
+
+public class AccountAccessBodyValidator implements ConstraintValidator<ValidConsentBody, AisConsentInitiateBody.AccountAccessBody> {
+
+    @Override
+    public void initialize(ValidConsentBody constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(AisConsentInitiateBody.AccountAccessBody body, ConstraintValidatorContext context) {
+        if (body == null) {
+            return true;
+        }
+
+        return isValidAisConsentStructure(body);
+    }
+
+    private boolean isValidAisConsentStructure(AisConsentInitiateBody.AccountAccessBody body) {
+        return isValidDedicatedConsent(body) || isValidGlobalConsent(body) || isValidBankOfferedConsent(body);
+    }
+
+    private boolean isValidDedicatedConsent(AisConsentInitiateBody.AccountAccessBody body) {
+        boolean validDedicatedWithAccounts = !isEmptyAccountInfo(body)
+                                                     && StringUtils.isEmpty(body.getAllPsd2())
```

Comment:
Suggest to use Guava Strings.isNullOrEmpty as the validator can be reused outside of Spring

Reply:
Guava Strings.isNullOrEmpty was used instead